pkg/system/phase4_configuring.go

```diff
@@ -364,6 +364,21 @@ func (r *Reconciler) SetDesiredDeploymentEndpoint() error {
 				r.setDesiredCoreEnv(c)
 			}
 
+			// Filter out NOOBAA_ROOT_SECRET from env vars to avoid exposing it in pod spec
+			// it is set via mounting the secret as files
+			// this will remove the leftover env var in case of an upgrade from older operator version (older then 4.21)
+			// as we preserve env vars on updates by merging the arrays and not replacing them.
+
+			if len(c.Env) > 0 {
```

Comment:
this is not necessary since `setDesiredCoreEnv` is used for the endpoints (line 364 above)